dotnet/src/SemanticKernel.Abstractions/AI/FunctionChoiceBehaviors/FunctionChoiceBehaviorOptions.cs

```diff
@@ -26,4 +26,13 @@ public sealed class FunctionChoiceBehaviorOptions
     /// </remarks>
     [JsonPropertyName("allow_concurrent_invocation")]
     public bool AllowConcurrentInvocation { get; set; } = false;
+
+    /// <summary>
+    /// Gets or sets whether the AI model should strictly adhere to the function schema.
+    /// </summary>
+    /// <remarks>
+    /// The default value is set to false. If set to true, the AI model will strictly adhere to the function schema.
+    /// </remarks>
+    [JsonPropertyName("allow_strict_schema_adherence")]
+    public bool AllowStrictSchemaAdherence { get; set; } = false;
```

Comment:
nit: `EnableStrictSchema` name sounds a little bit easier. I think the name of this property shouldn't necessarily align to other properties in this class, which starts with `Allow...`.

Reply:
I brought the exam same point to @SergeyMenshykh during our recent meeting. His point was that not all models might support strict mode, so it'll really depend on the connector implementation in the end. Hence the allow naming and not enable.

Reply:
do you have any guidance regarding the code?

Reply:
> His point was that not all models might support strict mode, so it'll really depend on the connector implementation in the end

That makes sense.

Reply:
> do you have any guidance regarding the code?

```csharp
[Experimental("SKEXP0001")]
[JsonPropertyName("allow_strict_schema_adherence")]
public bool AllowStrictSchemaAdherence { get; set; } = false;
```

Reply:
```suggestion
    [Experimental("SKEXP0001")]
    public bool AllowStrictSchemaAdherence { get; set; } = false;
```

Reply:
@baywet we always make new public API surface experimental to start with.

Please add an `[Experimental("SKEXP0010")]` attribute.